### PR TITLE
feat(compiler-cli): ngtsc as a ts_library plugin

### PR DESCRIPTION
**Pull request by alexeagle**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: packages/compiler-cli/index.ts [hunk 1]
```diff
@@ -24,3 +24,4 @@ export {CompilerOptions as AngularCompilerOptions} from './src/transformers/api'
 export {NgTools_InternalApi_NG_2 as __NGTOOLS_PRIVATE_API_2} from './src/ngtools_api';
 
 export {ngToTsDiagnostic} from './src/transformers/util';
+export {NgTscPlugin} from './src/ngtsc/tsc_plugin/tsc_plugin';
```
Comment (alxhub) [inline packages/compiler-cli/index.ts]: I think we should put this as `src/ngtsc/tsc_plugin.ts`, without the intervening directory. The convention in `compiler-cli` is that directories under `src/ngtsc` are distinct compilation units.

**Review by alxhub — APPROVED** [review]

File: packages/compiler-cli/index.ts (hunk 1 above)
Comment (alexeagle) [reply packages/compiler-cli/index.ts]: okay, moved moth of these new files to the `src/ngtsc` directory

**Comment by alexeagle** [comment]
Caretaker: the aio test failure appears to be because this PR is coming from a branch on the upstream rather than from a fork. It's unrelated - the code added in this PR isn't even used yet. Merge anyway please :)

**Comment by jasonaden** [comment]
@alexeagle I don't mind merging, but I don't see a `g3` presubmit here and `google3` is still marked `pending`.

**Review by IgorMinar — APPROVED** [review]

**Pull request closed without merging**